### PR TITLE
Add support for extra HTTP headers for "spacewalk-repo-sync" on RPM repos

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR enables `spacewalk-repo-sync` to use extra HTTP headers when dealing with RPM repositories:

- Extra HTTP headers can be now defined at `/etc/rhn/spacewalk-repo-sync/extra_headers.conf` file, in a similar way as you can set HTTP proxy settings:

```yaml
# You can define extra headers per channel label
[channel_label]
header1=value1
header2=value2

# Or by repository name
[repository_name]
another_header1=value
another_header2=value

# Or globally for all respositories and channels
[main]
a_global_header=value
```
- A new Zypper plugin called `spacewalk-extra-http-headers` has been also introduced in order to allow injecting the customer headers into the requests made by Zypper to retrieve the repository metadata.

This is particularly useful to deal with RHUI repositories within the public clouds.

**NOTE:** This PR also adds missing `ulnauth` and `spacewalk-uln-resolver` files to `spacewalk-backend-tools` package.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls) (will be linked shortly)

- [x] **DONE**

## Test coverage
- No tests: **no tests for reposync with extra headers**

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/10936

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by meaksh** [comment]
I'll be creating the corresponding issue for the documentation in a moment

File: backend/satellite_tools/spacewalk-extra-http-headers [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
```
Comment (mcalmer) [inline backend/satellite_tools/spacewalk-extra-http-headers:1]: Is the shebang changed when building the package? If not, I fear that using `env` is not allowed and we do not want python2 anymore.

Comment (meaksh) [reply backend/satellite_tools/spacewalk-extra-http-headers:1]: Yep, the shebang is changed when building `spacewalk-backend-tools`: https://github.com/uyuni-project/uyuni/blob/master/backend/spacewalk-backend.spec#L279

Comment (mcalmer) [reply backend/satellite_tools/spacewalk-extra-http-headers:1]: ok

**Pull request merged**